Session3/EFCore - Assignment 3 Base/EFCoreAssignment.Services/ProductService.cs

```diff
@@ -0,0 +1,77 @@
+﻿using EFCoreAssignment.Data.Entities;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Storage.ValueConversion.Internal;
+
+namespace EFCoreAssignment.Data.Services
+{
+    public class ProductService : IProductService
+    {
+        private readonly AppDbContext _context;
+
+        public ProductService(AppDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<List<ProductDto>> GetProducts()
+        {
+            // TODO get products
+            var product = await _context.Products.ToListAsync();
```

Comment:
You can chain .Select method after '_context.Products' then call the execute command. In this way we can be sure to run the command in the database and not in the client.

Reply:
hello here is the updated code
![image](https://github.com/miguelrefuerzo/EFCoreMasters/assets/106952123/5840b870-34e5-4e0f-9ee2-e0b477dd6da9)